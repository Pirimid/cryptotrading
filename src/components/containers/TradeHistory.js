import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {Grid, Row, Col } from 'react-bootstrap';
import * as actions from '../../actions/fuelSavingsActions';

export class TradeHistory extends React.Component {

  render() {
		return (
		  <div className="card card-fh">
		  	<div className="card-fixed-header cw2">
					<h2 className="card-title">Trade History</h2>
					<div className="card-panel-header">
						<label>Total Size</label>
						<label>Price(EUR)</label>
						<label>Time</label>
					</div>
				</div>
				<div className="card-panel-body m-t-fh">
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span className="up">9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span className="up">9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span className="up">9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span className="up">9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span className="up">9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span className="up">9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span className="up">9830.50</span><span>12:13:37</span>
					</div><div className="card-panel-body-rows">
						<span>0.01879837</span><span className="up">9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span className="down">9830.17</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span className="down">9830.17</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span className="down">9830.17</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span className="down">9830.17</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span className="down">9830.17</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span className="down">9830.17</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span className="down">9830.17</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>

					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div><div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>

					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div><div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>

					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
					<div className="card-panel-body-rows">
						<span>0.01879837</span><span>9830.50</span><span>12:13:37</span>
					</div>
				</div>
			</div>
		);
  }
}

TradeHistory.propTypes = {
  actions: PropTypes.object.isRequired,
  currentPair: PropTypes.object.isRequired
};

function mapStateToProps(state) {
  return {
	currentPair: state.currentPair
  };
}

function mapDispatchToProps(dispatch) {
  return {
	actions: bindActionCreators(actions, dispatch)
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TradeHistory);
