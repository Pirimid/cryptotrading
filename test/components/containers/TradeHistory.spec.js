import React from "react";
import { mount, shallow } from "enzyme";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import { create } from "react-test-renderer";
import TradeHistory from "../../../src/components/containers/TradeHistory";
import initialState from "../../../src/reducers/initialState";

describe("<TradeHistory />", () => {
  const store = configureMockStore()(initialState);

  it("should match snapshot", () => {
    const component = create(
      <Provider store={store}>
        <TradeHistory />
      </Provider>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

});