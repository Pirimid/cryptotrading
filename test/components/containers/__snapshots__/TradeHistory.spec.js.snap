// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TradeHistory /> should match snapshot 1`] = `
<div
  className="card card-fh"
>
  <div
    className="card-fixed-header"
  >
    <h2
      className="card-title"
      data-card="tradehistory"
      onClick={undefined}
    >
      Trade History
    </h2>
    <div
      className="card-panel-header col-w-4"
    >
      <label>
        Total Size
      </label>
      <label>
        Price(
        USD
        )
      </label>
      <label>
        Time
      </label>
      <label>
        Exchange
      </label>
    </div>
  </div>
  <div
    className="card-panel-body col-w-4"
  >
    <div
      className="card-panel-body-rows"
    >
      <span>
        0.02279837
      </span>
      <span
        className="up arrow"
      >
        6574.529651
      </span>
      <span>
        22:28:54
      </span>
      <span>
        GDAX
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span>
        0.01879
      </span>
      <span
        className="up arrow"
      >
        6574.529451
      </span>
      <span>
        22:29:50
      </span>
      <span>
        GDAX
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span>
        5.05579837
      </span>
      <span
        className="up arrow"
      >
        6599.642222
      </span>
      <span>
        22:28:50
      </span>
      <span>
        Bitfinex
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span>
        0.02279837
      </span>
      <span
        className="down arrow"
      >
        6564.244552
      </span>
      <span>
        22:27:50
      </span>
      <span>
        Bittrex
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span>
        0.05579837
      </span>
      <span
        className="down arrow"
      >
        6599.642542
      </span>
      <span>
        22:27:49
      </span>
      <span>
        Bitfinex
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span>
        0.02279837
      </span>
      <span
        className="up arrow"
      >
        6561.451276
      </span>
      <span>
        22:26:40
      </span>
      <span>
        CEX.IO
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span>
        0.01879
      </span>
      <span
        className="down arrow"
      >
        6566.952425
      </span>
      <span>
        22:27:56
      </span>
      <span>
        Binance
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span>
        0.01879
      </span>
      <span
        className="down arrow"
      >
        6576.052516
      </span>
      <span>
        22:26:50
      </span>
      <span>
        Kraken
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span>
        0.05579837
      </span>
      <span
        className="up arrow"
      >
        6574.529541
      </span>
      <span>
        22:26:30
      </span>
      <span>
        GDAX
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span>
        0.01279837
      </span>
      <span
        className="up arrow"
      >
        6599.646542
      </span>
      <span>
        22:26:20
      </span>
      <span>
        Bitfinex
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span>
        0.13879837
      </span>
      <span
        className="down arrow"
      >
        6615.514237
      </span>
      <span>
        22:25:47
      </span>
      <span>
        HitBTC
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span>
        0.01279837
      </span>
      <span
        className="down arrow"
      >
        6564.246252
      </span>
      <span>
        22:25:48
      </span>
      <span>
        Bittrex
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span>
        0.02279837
      </span>
      <span
        className="up arrow"
      >
        6574.529251
      </span>
      <span>
        22:22:50
      </span>
      <span>
        GDAX
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span>
        4.13879837
      </span>
      <span
        className="up arrow"
      >
        6576.054716
      </span>
      <span>
        22:25:50
      </span>
      <span>
        Kraken
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span>
        1.01279837
      </span>
      <span
        className="down arrow"
      >
        6574.512651
      </span>
      <span>
        22:25:41
      </span>
      <span>
        GDAX
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span>
        0.01279837
      </span>
      <span
        className="up arrow"
      >
        6599.643642
      </span>
      <span>
        22:21:10
      </span>
      <span>
        Bitfinex
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span>
        0.05579837
      </span>
      <span
        className="up arrow"
      >
        6576.054721
      </span>
      <span>
        22:21:50
      </span>
      <span>
        GDAX
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span>
        1.11879
      </span>
      <span
        className="up arrow"
      >
        6599.646547
      </span>
      <span>
        22:20:30
      </span>
      <span>
        Bitfinex
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span>
        0.01879
      </span>
      <span
        className="up arrow"
      >
        6566.952145
      </span>
      <span>
        22:25:30
      </span>
      <span>
        Binance
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span>
        0.01279837
      </span>
      <span
        className="down arrow"
      >
        6564.241252
      </span>
      <span>
        22:21:01
      </span>
      <span>
        Bittrex
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span>
        0.13879837
      </span>
      <span
        className="up arrow"
      >
        6576.051476
      </span>
      <span>
        22:19:06
      </span>
      <span>
        Kraken
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span>
        0.02279837
      </span>
      <span
        className="down arrow"
      >
        6576.054121
      </span>
      <span>
        22:20:07
      </span>
      <span>
        GDAX
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span>
        0.13879837
      </span>
      <span
        className="up arrow"
      >
        6576.059636
      </span>
      <span>
        22:21:06
      </span>
      <span>
        Kraken
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span>
        9.91279837
      </span>
      <span
        className="up arrow"
      >
        6599.646123
      </span>
      <span>
        22:19:10
      </span>
      <span>
        Bitfinex
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span>
        0.01279837
      </span>
      <span
        className="down arrow"
      >
        6564.241245
      </span>
      <span>
        22:19:01
      </span>
      <span>
        Bittrex
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span>
        0.15579837
      </span>
      <span
        className="up arrow"
      >
        6614.522367
      </span>
      <span>
        22:19:50
      </span>
      <span>
        HitBTC
      </span>
    </div>
  </div>
</div>
`;
